src/hotspot/share/runtime/arguments.cpp

```diff
@@ -2952,6 +2955,37 @@ jint Arguments::parse_each_vm_init_arg(const JavaVMInitArgs* args, bool* patch_m
     LogConfiguration::configure_stdout(LogLevel::Info, true, LOG_TAGS(class, path));
   }
 
+  // This must be done after all options have been processed
+  // since -Xlog and -Xloggc override corresponding -verbose
+  for (LogTagSet* ts = LogTagSet::first(); ts != NULL; ts = ts->next()) {
+    if (ts->contains(LogTag::_gc) && ts->has_output_to_file()) {
+      has_verbose_gc = false;
+    }
```

Comment:
What if the command-line is this?

```
java -Xlog:gc+heap=foo.txt -verbose:gc -version
```

Reply:
According to the current commit, we will have 
```
gc -> disabled
gc+heap ->foo.txt
```
But the same as the comment above, I'm not sure about its correctness. So maybe we should discuss what if the selections in Xlog is not the same as ``-Xlog:gc``.